includes/Core/REST_API/Exception/Invalid_Dimensions_Exception.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\REST_API\Exception\Invalid_Dimensions_Exception
+ *
+ * @package   Google\Site_Kit\Core\REST_API\Exception
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\REST_API\Exception;
+
+use Google\Site_Kit\Core\Contracts\WP_Errorable;
+use Exception;
+use WP_Error;
+
+/**
+ * Exception thrown when dimensions are invalid for a report request.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Invalid_Dimensions_Exception extends Exception implements WP_Errorable {
```

Comment:
Let's extend PHP's [`DomainException`](https://www.php.net/manual/en/class.domainexception.php) as a more appropriate base instead of the top-level exception.